packages/js/product-editor/src/hooks/test/use-error-handler.test.ts

```diff
@@ -108,11 +108,70 @@ describe( 'useErrorHandler', () => {
 			errorProps.actions[ 0 ].onClick();
 		}
 
-		jest.runAllTimers(); // Run all pending timers (for setTimeout)
+		expect( mockFocusByValidatorId ).toHaveBeenCalledWith( 'sku' );
+	} );
+	it( 'should call getParentTabIdByBlockName and focusByValidatorId for product_invalid_sku error when errorProps action is triggered', () => {
+		const error = {
+			code: 'product_form_field_error',
+			validatorId: 'test-validator',
+		} as WPError;
+		const visibleTab = 'general';
+
+		jest.useFakeTimers();
+
+		const { result } = renderHook( () => useErrorHandler() );
+		const { getProductErrorMessageAndProps } = result.current;
+
+		const { errorProps: fieldsErrorProps } = getProductErrorMessageAndProps(
+			error,
+			visibleTab
+		);
+
+		expect( fieldsErrorProps ).toBeDefined();
+		expect( fieldsErrorProps.actions ).toBeDefined();
+		expect( fieldsErrorProps.actions?.length ).toBeGreaterThan( 0 );
+
+		// Trigger the action
+		if ( fieldsErrorProps.actions && fieldsErrorProps.actions.length > 0 ) {
+			fieldsErrorProps.actions[ 0 ].onClick();
+		}
 
-		expect( mockNavigateTo ).toHaveBeenCalledWith( { url: '/new-path' } );
 		expect( mockFocusByValidatorId ).toHaveBeenCalledWith(
 			'test-validator'
 		);
 	} );
+	it( 'should call getParentTabIdByBlockName for product_invalid_sku error when errorProps action is triggered and does not find sku block name', () => {
+		const errors = {
+			code: 'product_invalid_sku',
+		} as WPError;
+		const visibleTab = 'general';
+
+		jest.mock( '../use-blocks-helper', () => ( {
+			useBlocksHelper: jest.fn().mockReturnValue( {
+				getParentTabId: jest.fn( () => 'inventory' ),
+				getParentTabIdByBlockName: jest.fn( () => null ), // Mock returns null
+			} ),
+		} ) );
+
+		jest.useFakeTimers();
+
+		const { result } = renderHook( () => useErrorHandler() );
+		const { getProductErrorMessageAndProps } = result.current;
+
+		const { errorProps } = getProductErrorMessageAndProps(
+			errors,
+			visibleTab
+		);
+
+		expect( errorProps ).toBeDefined();
+		expect( errorProps.actions ).toBeDefined();
+		expect( errorProps.actions?.length ).toBeGreaterThan( 0 );
+
+		// Trigger the action
+		if ( errorProps.actions && errorProps.actions.length > 0 ) {
+			errorProps.actions[ 0 ].onClick();
+		}
+
+		expect( mockFocusByValidatorId ).toHaveBeenCalledWith( 'sku' );
```

Comment:
We should also be testing the call to `navigateTo` here, because in this instance `navigateTo` will be called with a `tab=null` param which would be invalid.
If the tab id is not found I don't think we should be returning any actions to begin with as we wouldn't know where to direct a user to.
This looks like it highlighted a bug in our logic.

Reply:
Good call! I added a check to the different cases (in the commit [118ca65](https://github.com/woocommerce/woocommerce/pull/49729/commits/118ca6530fd5a079fd8ef7efa9c93728e54ca553)) to avoid calling `navigateTo` with `null`. After that check, I think it won't be necessary to test the `navigateTo` with `null`.